modules/phase_field/src/postprocessors/GrainTracker.C

```diff
@@ -684,19 +677,19 @@ GrainTracker::attemptGrainRenumber(MooseSharedPointer<FeatureData> grain, unsign
 
     // If we reach this part of the loop, there is no simple renumbering that can be done.
     mooseAssert(_unique_grains.find(target_it->_grain_id) != _unique_grains.end(), "Error in indexing target grain in attemptGrainRenumber");
-    MooseSharedPointer<FeatureData> target_grain = _unique_grains[target_it->_grain_id];
+    FeatureData & target_grain = *_unique_grains[target_it->_grain_id];
```

Comment:
Possibly a good point to use an `auto &` as a return value... it would help insulate this line of code from changing in the future (if you change how the data is stored again)